azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CAutoConfiguration.java

```diff
@@ -0,0 +1,95 @@
+ * license information.
+ */
+package com.microsoft.azure.spring.autoconfigure.b2c;
+
+import lombok.NonNull;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.client.registration.InMemoryClientRegistrationRepository;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizationRequestResolver;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+import org.springframework.security.oauth2.core.ClientAuthenticationMethod;
+
+import java.util.UUID;
+
+import static com.microsoft.azure.spring.autoconfigure.b2c.AADB2CProperties.*;
+
+@Configuration
+@ConditionalOnWebApplication
+@ConditionalOnProperty(
+        prefix = PREFIX,
+        value = {
+                "tenant",
+                "client-id",
+                "client-secret",
+                POLICY_SIGN_UP_OR_SIGN_IN_NAME,
+                POLICY_SIGN_UP_OR_SIGN_IN_REPLY_URL
+        }
+)
+@EnableConfigurationProperties(AADB2CProperties.class)
+public class AADB2CAutoConfiguration {
+
+    private static final String CLIENT_NAME = "aad-b2c";
+
+    private static final String REGISTRATION_ID = "aad-b2c-" + UUID.randomUUID();
+
+    private final ClientRegistrationRepository repository;
+
+    private final AADB2CProperties properties;
+
+    public AADB2CAutoConfiguration(@NonNull ClientRegistrationRepository repository,
+                                   @NonNull AADB2CProperties properties) {
+        this.repository = repository;
+        this.properties = properties;
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public OAuth2AuthorizationRequestResolver b2cOAuth2AuthorizationRequestResolver() {
+        return new AADB2CAuthorizationRequestResolver(repository, properties);
+    }
+
+    @Configuration
+    @ConditionalOnProperty(prefix = PREFIX, value = "oidc-enabled", havingValue = "true", matchIfMissing = true)
+    public static class AADB2COidcAutoConfiguration {
+
+        private final AADB2CProperties properties;
+
+        public AADB2COidcAutoConfiguration(@NonNull AADB2CProperties properties) {
+            this.properties = properties;
+        }
+
+        @Bean
+        @ConditionalOnMissingBean
+        public ClientRegistrationRepository clientRegistrationRepository() {
+            return new InMemoryClientRegistrationRepository(this.b2cOIDCClientRegistration());
```

Comment:
Does that mean there will always be only one ClientRegistration available for the user? i.e., B2C here. If it is, do we need to consider support multiple client registrations? e.g., user configured other third party OAuth2 providers.

Reply:
good catch, we should add condititonalon b2c properties here.

Reply:
Sorry, this bean is under conditionProperty already. See line 62. If user would like to leverage other providers, just remove the property `azure.activedirectory.b2c`.

Reply:
For multi provider support, we may file one issue for this. And will cover that later.